src/stormpath.login.js

```diff
@@ -2,7 +2,12 @@
 
 angular.module('stormpath')
 
-.controller('SpLoginFormCtrl', ['$scope','$auth',function ($scope,$auth) {
+.controller('SpLoginFormCtrl', ['$scope','$auth','$socialLogin',function ($scope,$auth,$socialLogin) {
+  // Load list of social login providers from server.
+  $socialLogin.getProviders().then(function(providers) {
```

Comment:
Because of the network request, the provider buttons won't appear immediately (I think this is a bad experience).  I think that the `getProviders` function should cache the HTTP result.  If I want that resource to be pre-fetched (before the login form is rendered) I could put a call to `getProviders` in a `app.run()` block